### ISSUE TITLE
:wrench: Implement a quantum mixer :mage: 

### ISSUE THREAD

**Issue by paxcema** [mindsdb/lightwood#641]
# :wrench: Implement a quantum mixer :mage: 

## Introduction

Lightwood is an AutoML library that aims at abstracting common procedures in the Machine Learning usual workflow (e.g. data preparation, feature engineering, model building, model analysis) with an approach that can be fully automated but easily customizable and extended thanks to its flexibility. 

One of the key tenets of Lightwood's philosophy is to engineer features by using _encoders_ that might transform columnar data into a useful descriptor. These are then fed into what is called a _mixer_: the actual ML model tasked with learning to predict the specified target. 

This approach is very flexible. Recently, we've come across the great [Qiskit SDK](https://qiskit.org/) for working with quantum computers, and thought it could be very cool to experiment with _hybrid quantum-classical neural networks_ as mixers! 

## Task

Implement a proof-of-concept `QuantumMixer` class that inherits from our `BaseMixer` class and uses a hybrid quantum neural network as the learning mechanism. We suggest closely following this [Qiskit tutorial](https://qiskit.org/textbook/ch-machine-learning/machine-learning-qiskit-pytorch.html) for an initial implementation.

A good way to measure success is to implement a unit test that uses the quantum mixer in a simple task. For example, you can use [this integration test](https://github.com/mindsdb/lightwood/blob/46ba63adc9037bda45f0c864e72ec1da4029512e/tests/integration/basic/test_airline.py#L8) as reference and opt for the [HDI](https://github.com/mindsdb/lightwood/blob/staging/tests/data/hdi.csv) toy dataset. Be sure to read the [custom mixer tutorial](https://lightwood.io/tutorials/custom_mixer/custom_mixer.html) so that you can force quantum mixer usage in the test.

> Note: mixer accuracy is not relevant at this time, future work can tackle it. Additionally, you can opt for testing with synthetic data if that's easier for you.

The core Lightwood team would love to be of assistance to any contributors interested in tackling this challenge. If at any point you have questions and/or want to discuss relevant details, feel free to post them here or in the community slack.

## Steps :male_detective: :female_detective: 

- Fork the Lightwood repository, checkout the `staging` branch and from it create a new one.
- Implement the `QuantumMixer` class, using all resources suggested above. Any files should be added inside the `lightwood/encoder/quantum` directory.
- Implement a unit test where your mixer is successfully used with a sample dataset.
- Make the PR and address any comments that reviewers might make.

## Additional rewards :1st_place_medal: 

This challenge encompasses mixer research. As such, an accepted PR brings :one::zero: points for entry into the draw for a :computer:  Deep Learning Laptop powered by the NVIDIA RTX 3080 Max-Q GPU or other swag :shirt: :bear: .  For more info check out https://mindsdb.com/hacktoberfest/

**Comment by ongspxm** [comment]
Hi saw that there alr is a pr, but hope to work on this qiskit implementation too, I a have slightly different implementation that kinda builds on default net, will it be considered duplicated effort?

**Comment by mrandri19** [comment]
> Hi saw that there alr is a pr, but hope to work on this qiskit implementation too, I a have slightly different implementation that kinda builds on default net, will it be considered duplicated effort?

Hey, I am curious about you implementation too so I suppose you could open a PR as well. We could even join our efforts and pick the best parts of both our solutions. I suppose splitting the points would not be a problem for them at all.
